Classes/Domain/Model/Changes/CopyInto.php

```diff
@@ -73,8 +73,16 @@ public function getMode()
     public function apply()
     {
         if ($this->canApply()) {
-            $nodeName = $this->generateUniqueNodeName($this->getParentNode());
-            $node = $this->getSubject()->copyInto($this->getParentNode(), $nodeName);
+            $parentNode = $this->getParentNode();
+            $nodeName = $this->generateUniqueNodeName($parentNode);
+            // If the parent node has children, we copy the node after the last child node to prevent the copied nodes
+            // from being mixed with the existing ones due the duplication of their relative indices.
+            if ($parentNode->hasChildNodes()) {
+                $lastChildNode = array_slice($parentNode->getChildNodes(), -1, 1)[0];
+                $node = $this->getSubject()->copyAfter($lastChildNode, $nodeName);
+            } else {
+                $node = $this->getSubject()->copyInto($parentNode, $nodeName);
+            }
```

Comment:
that makes absolutely sense. But i will have to check how its done in Neos9 to align behaviour.